### DISCUSSION TITLE
nuxt components slot props typing ?

### DISCUSSION THREAD

**Discussion by germsb** [nuxt/nuxt#29105] [Questions]
Hello,

Does anyone have an idea how to enable autocomplete for slot props of Nuxt components, as well as those from modules like `'@nuxtjs/i18n', '@nuxt/ui'`

Otherwise, everything works as it should.

Merci d'avance :)

Ok for vue comp
![Capture d’écran du 2024-09-21 17-25-50](https://github.com/user-attachments/assets/5e211dab-2121-4b94-82a9-61e994f0a19f)

Not for Nuxt comp
![Capture d’écran du 2024-09-21 17-26-39](https://github.com/user-attachments/assets/952f5919-41e1-4822-837d-095014ec3ff1)

**Comment by JulianWowra** [comment]
If you build your own components, you must specify the types using [`defineSlots<>()`](https://vuejs.org/api/sfc-script-setup.html#defineslots)

```vue
<template>
	<div>
		<slot name="mySlot" :myTypedProp="sampleRef"></slot>
	</div>
</template>

<script lang="ts" setup>
const sampleRef = ref('Your value will be passed');

const slots = defineSlots<{
  mySlot(props: { myTypedProp: string }): unknown
}>();
</script>
```

If `@nuxt/ui` have not done this in their components, then you can only make an issue or a pull request

**Reply by germsb** [reply]
Can you test if intelisense on NuxtLink template work for you ?

**Reply by JulianWowra** [reply]
Same issue as you

**Comment by germsb** [comment]
Sure, thanks for the answer,
What I mean is, in vscode, volar does not provide intellisense for the components mentioned above.
As for Nuxt UI, since I upgraded to v3, I have no more issues.
